lib/cublas/wrappers.jl

```diff
@@ -126,17 +126,29 @@ for (fname, fname_64, elty) in ((:cublasDscal_v2, :cublasDscal_v2_64, :Float64),
         end
     end
 end
+function scal!(n::Integer, alpha::Number, x::StridedCuVecOrDenseMat{T}) where {T}
+    α = convert(T, alpha)
+    gpu_α = CuRef{T}(α) 
+    scal!(n, gpu_α, x)
+    synchronize()
+    x
+end
+function scal!(n::Integer, alpha::CuRefArray{Float32, CuVector{Float32, DeviceMemory}}, x::StridedCuVecOrDenseMat{Float16})
+    cublasScalEx(handle(), n, alpha, Float32, x, Float16, stride(x, 1), Float32)
+    return x
+end
 function scal!(n::Integer, alpha::Number, x::StridedCuVecOrDenseMat{Float16})
-    α = convert(Float32, alpha)
-    cublasScalEx(handle(), n, Ref{Float32}(α), Float32, x, Float16, stride(x, 1), Float32)
+    α = CuRef{Float32}( convert(Float32, alpha) )
```

Comment:
We should improve `CuRef` so that it can be constructed identically to `Ref`. `Ref{T}(x)` doing an implicit `convert` is pretty convenient.